src/cc/libbpf.c

```diff
@@ -856,47 +856,100 @@ static int bpf_attach_tracing_event(int progfd, const char *event_path, int pid,
   return 0;
 }
 
+static int create_kprobe_event(char *buf, const char *ev_name,
+                               enum bpf_probe_attach_type attach_type,
+                               const char *fn_name, uint64_t fn_offset,
+                               int maxactive)
+{
+  int kfd;
+  char ev_alias[128];
+  static unsigned int buf_size = 256;
+
+  kfd = open("/sys/kernel/debug/tracing/kprobe_events", O_WRONLY | O_APPEND, 0);
+  if (kfd < 0) {
+    fprintf(stderr, "open(/sys/kernel/debug/tracing/kprobe_events): %s\n", buf,
+            strerror(errno));
+    return -1;
+  }
+
+  snprintf(ev_alias, sizeof(ev_alias), "%s_bcc_%d", ev_name, getpid());
+
+  if (fn_offset > 0 && attach_type == BPF_PROBE_ENTRY)
+    snprintf(buf, buf_size, "p:kprobes/%s %s+%"PRIu64,
+             ev_alias, fn_name, fn_offset);
+  else if (maxactive > 0 && attach_type == BPF_PROBE_RETURN)
+    snprintf(buf, buf_size, "r%d:kprobes/%s %s",
+             maxactive, ev_alias, fn_name);
+  else
```

Comment:
Does `maxactive` do anything in k(entry)probe? If not, maybe we should check that the value should be 0?

Reply:
`maxactive` doesn't do anything for kprobes. Do you mean check that the value is 0 at the beginning of `bpf_attach_kprobe`?